subprojects/plugins/src/integTest/groovy/org/gradle/groovy/compile/GroovyCompileToolchainIntegrationTest.groovy

```diff
@@ -0,0 +1,211 @@
+/*
+ * Copyright 2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.groovy.compile
+
+import org.gradle.api.JavaVersion
+import org.gradle.integtests.fixtures.AbstractIntegrationSpec
+import org.gradle.integtests.fixtures.AvailableJavaHomes
+import org.gradle.integtests.fixtures.jvm.JavaToolchainFixture
+import org.gradle.internal.jvm.Jvm
+import org.gradle.test.fixtures.file.TestFile
+import org.gradle.util.internal.TextUtil
+
+import static org.gradle.util.internal.GroovyDependencyUtil.groovyModuleDependency
+
+class GroovyCompileToolchainIntegrationTest extends AbstractIntegrationSpec implements JavaToolchainFixture {
+
+    // The following versions and values must be kept in sync
+    def groovyVersion = "3.0.13"
```

Comment:
❌ Would it make sense that this becomes a multiversion test? We also have quite some information in `GroovyCoverage`, I suppose we should refer to that instead of re-declaring things here.

Reply:
Using `GroovyCoverage` is indeed better. It should also help with testing newer Groovy version when they are added to the coverage.

I turned the integration test into a multi-version test. It now runs the tests against all Groovy 3 and Groovy 4 versions specified in the coverage and supported by the JVM executing the tests.